upf/transformers/conditional_effects_remover.py

```diff
@@ -63,23 +62,23 @@ def get_rewritten_problem(self) -> Problem:
                         em = self._env.expression_manager
                         c = e.condition()
                         nv = self._simplifier.simplify(em.Or(em.And(c, v), em.And(em.Not(c), f)))
-                        self._new_problem.add_timed_effect(t, f, nv)
+                        self._new_problem.add_timed_effect(t, e.fluent(), nv)
                 else:
-                    self._new_problem.add_timed_effect(t, e)
-        self._new_problem_add_timed_goals()
-        self._new_problem_add_maintain_goals()
-        self._new_problem_add_goals()
+                    self._new_problem._add_effect_instance(t, e.clone())
+        self._new_problem.clear_actions()
         for ua in self._problem.unconditional_actions():
             self._new_problem.add_action(ua)
```

Comment:
you need to clone the action here

Reply:
@Framba-Luca double check in all the transformers that we clone all the mutable objects before adding them to the new problem.